pkg/cmd/ssh.go

```diff
@@ -279,22 +293,64 @@ func getNodeNameForIP(ip string) string {
 	checkError(err)
 	for _, node := range nodes.Items {
 		if ip == node.Status.Addresses[0].Address {
-			return node.Name
+			return &node
 		}
 	}
-	return ""
+	return nil
 }
 
 // fetchAWSImageIDByInstancePrivateIP returns the image ID (AMI) for instance with the given <privateIP>.
 func fetchAWSImageIDByInstancePrivateIP(privateIP string) (string, error) {
-	command := fmt.Sprintf("aws ec2 describe-instances --filters Name=private-ip-address,Values=%s --query Reservations[0].Instances[0].ImageId", privateIP)
-	captured := capture()
-	operate("aws", command)
-	output, err := captured()
+	clientToTarget("garden")
+	checkError(err)
+	clientset, err := v1alpha1.NewForConfig(NewConfigFromBytes(*kubeconfig))
+	checkError(err)
+	controllerRegistration, err := clientset.ControllerRegistrations().Get(ControllerRegistrationAwsName, metav1.GetOptions{})
+	checkError(err)
 
-	if err != nil || output == "None" {
-		return "", fmt.Errorf("could not fetch image ID (AMI) of instance with private IP address %q", privateIP)
+	//var data string
+	var controllerRegistrationSpec AWSControllerRegistrationSpec
+	err = json.Unmarshal(controllerRegistration.Spec.Deployment.ProviderConfig.Raw, &controllerRegistrationSpec)
+	checkError(err)
+	machineImages := controllerRegistrationSpec.Values.Config.MachineImages
+
+	nodeRegion := getNodeForIP(privateIP).Labels[NodeRegionLabel]
+	var ami string
+	for _, machineImage := range machineImages {
+		if machineImage.Name == BastionOSName {
+			for _, region := range machineImage.Regions {
+				if region.Name == nodeRegion {
+					ami = region.Ami
+					break
+				}
+			}
+		}
+	}
+
+	if ami == "" {
+		fmt.Println(fmt.Sprintf("Cannot find AMI with coreos image for region: %s", nodeRegion))
 	}
 
-	return output, nil
+	return ami, nil
```

Comment:
if there is no case were an error is returned we can remove the error return value